WalletWasabi.Fluent/Helpers/LinuxStartupHelper.cs

```diff
@@ -7,35 +7,52 @@ namespace WalletWasabi.Fluent.Helpers
 {
 	public static class LinuxStartupHelper
 	{
+		private static readonly string DesktopFileContent = string.Join(
+					"\n",
+					"[Desktop Entry]",
+					$"Name={Constants.AppName}",
+					"Type=Application",
+					$"Exec={EnvironmentHelpers.GetExecutablePath()}",
+					"Hidden=false",
+					"Terminal=false",
+					"X-GNOME-Autostart-enabled=true");
+
+		private static string PathToDir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".config", "autostart");
+		private static string PathToDesktopFile = Path.Combine(PathToDir, "Wasabi.desktop");
+
 		public static async Task AddOrRemoveDesktopFileAsync(bool runOnSystemStartup)
 		{
-			string pathToDir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".config", "autostart");
-			string pathToDesktopFile = Path.Combine(pathToDir, "Wasabi.desktop");
-
-			IoHelpers.EnsureContainingDirectoryExists(pathToDesktopFile);
+			IoHelpers.EnsureContainingDirectoryExists(PathToDesktopFile);
 
 			if (runOnSystemStartup)
 			{
 				string pathToExec = EnvironmentHelpers.GetExecutablePath();
 
 				IoHelpers.EnsureFileExists(pathToExec);
 
-				string fileContents = string.Join(
-					"\n",
-					"[Desktop Entry]",
-					$"Name={Constants.AppName}",
-					"Type=Application",
-					$"Exec={pathToExec}",
-					"Hidden=false",
-					"Terminal=false",
-					"X-GNOME-Autostart-enabled=true");
-
-				await File.WriteAllTextAsync(pathToDesktopFile, fileContents).ConfigureAwait(false);
+				await File.WriteAllTextAsync(PathToDesktopFile, DesktopFileContent).ConfigureAwait(false);
 			}
 			else
 			{
-				File.Delete(pathToDesktopFile);
+				File.Delete(PathToDesktopFile);
 			}
 		}
+
+		internal static bool CheckDesktopFile()
+		{
+			bool result = false;
+			if (File.Exists(PathToDesktopFile))
+			{
+				result = CheckFileContent();
+			}
+			return result;
+		}
+
+		private static bool CheckFileContent()
+		{
+			string realFileContent = string.Join("\n", File.ReadAllLines(PathToDesktopFile));
```

Comment:
Async?

Reply:
AFAIK there is no async static method in `string`.

Reply:
That is true. I meant File.ReadAllLinesAsync

Reply:
Oh right. i totally forgot about that. Done.